tfx/components/base/base_executor.py

```diff
@@ -160,3 +160,116 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
          output_dict: Dict[Text, List[types.Artifact]],
          exec_properties: Dict[Text, Any]) -> None:
     pass
+
+
+class BeamExecutor(BaseExecutor):
+  """ Abstract TFX executor class for handling Beam-related logic."""
+
+  @abc.abstractmethod
+  def beam_io_signature(self, input_dict: Dict[Text, List[types.Artifact]],
+                        output_dict: Dict[Text, List[types.Artifact]],
+                        exec_properties: Dict[Text, Any]
+                        ) -> List[Dict[Text, Any]]:
+    """Provides input and output signatures for the input_dict and output_dict
+       of the underlying component.
+
+    Args:
+      input_dict: Input dict from input key to a list of Artifacts. These are
+        often outputs of another component in the pipeline and passed to the
+        component by the orchestration system.
+      output_dict: Output dict from output key to a list of Artifacts. These are
+        often consumed by a dependent component.
+      exec_properties: A dict of execution properties. These are inputs to
+        pipeline with primitive types (int, string, float) and fully
+        materialized when a pipeline is constructed. No dependency to other
+        component or later injection from orchestration systems is necessary or
+        possible on these values.
+
+    Returns:
+      A first dict from input artifact name/split key to the return type of
+      reading the inputs. A second dict from output artifact name/split key to
+      the return type of executing underlying component logic.
+    """
+    pass
+
+  @abc.abstractmethod
+  def read_inputs(self, pipeline: beam.Pipeline,
+                  input_dict: Dict[Text, List[types.Artifact]],
+                  output_dict: Dict[Text, List[types.Artifact]],
+                  exec_properties: Dict[Text, Any]
+                  ) -> Dict[Text, beam.pvalue.PCollection]:
+    """Executes reads for underlying component implementation.
+
+    Args:
+      input_dict: Input dict from input key to a list of Artifacts. These are
+        often outputs of another component in the pipeline and passed to the
+        component by the orchestration system.
+      output_dict: Output dict from output key to a list of Artifacts. These are
+        often consumed by a dependent component.
+      exec_properties: A dict of execution properties. These are inputs to
+        pipeline with primitive types (int, string, float) and fully
+        materialized when a pipeline is constructed. No dependency to other
+        component or later injection from orchestration systems is necessary or
+        possible on these values.
+
+    Returns:
+      A dict from input artifact name/split key to PCollections from reads for
+      underlying component execution logic.
+    """
+    pass
+
+  @abc.abstractmethod
+  def construct_beam_graph(self, pipeline: beam.Pipeline,
```

Comment:
@charlesccychen I'm not a fan of this name because it's not parallel with `read_inputs` and `write_outputs` - those two methods also construct Beam graphs, so why should this method have the claim to the name?

Something like `execute_component` or `run_component` or `process_artifacts` might make more sense?

Reply:
I changed the name to `run_component` to make it parallel. @charlesccychen let me know if this name is alright for you.